bot/seasons/easter/egg_hunt/cog.py

```diff
@@ -0,0 +1,511 @@
+import asyncio
+import contextlib
+import logging
+import random
+import sqlite3
+from datetime import datetime, timezone
+from pathlib import Path
+
+import discord
+from discord.ext import commands
+
+from bot.constants import Channels, Client, Roles as MainRoles, bot
+from bot.decorators import with_role
+from .constants import Colours, EggHuntSettings, Emoji, Roles
+
+log = logging.getLogger(__name__)
+
+DB_PATH = Path("bot", "resources", "persist", "egg_hunt.sqlite")
+
+TEAM_MAP = {
+    Roles.white: Emoji.egg_white,
+    Roles.blurple: Emoji.egg_blurple,
+    Emoji.egg_white: Roles.white,
+    Emoji.egg_blurple: Roles.blurple
+}
+
+GUILD = bot.get_guild(Client.guild)
+
+
+def get_team_role(user: discord.Member) -> discord.Role:
+    """Helper function to get the team role for a member."""
+
+    if Roles.white in user.roles:
+        return Roles.white
+    if Roles.blurple in user.roles:
+        return Roles.blurple
+
+
+async def assign_team(user: discord.Member) -> discord.Member:
+    """Helper function to assign a new team role for a member."""
+
+    db = sqlite3.connect(DB_PATH)
+    c = db.cursor()
+    c.execute(f"SELECT team FROM user_scores WHERE user_id = {user.id}")
```

Comment:
While the danger is pretty much non-existent, this should really be sanitized using sqlite's `?` syntax.

Reply:
I disagree on the excessive sanitation. There's no user provided values at all going into any SQL, so there's no vectors for them to manipulate it. It would negatively impact speed and readability for no gain.

Reply:
Alright that's fair, thought that may be the reason.